presto-main/src/main/java/com/facebook/presto/sql/planner/NodePartitioningManager.java

```diff
@@ -151,21 +165,30 @@ public NodePartitionMap getNodePartitioningMap(Session session, PartitioningHand
 
     public BucketNodeMap getBucketNodeMap(Session session, PartitioningHandle partitioningHandle, boolean preferDynamic)
     {
-        ConnectorBucketNodeMap connectorBucketNodeMap = getConnectorBucketNodeMap(session, partitioningHandle);
-
-        if (connectorBucketNodeMap.hasFixedMapping()) {
-            return new FixedBucketNodeMap(getSplitToBucket(session, partitioningHandle), getFixedMapping(connectorBucketNodeMap));
-        }
-
-        if (preferDynamic) {
-            return new DynamicBucketNodeMap(getSplitToBucket(session, partitioningHandle), connectorBucketNodeMap.getBucketCount());
+        ConnectorId connectorId = partitioningHandle.getConnectorId()
+                .orElseThrow(() -> new IllegalArgumentException("No connector ID for partitioning handle: " + partitioningHandle));
+        List<Node> sortedNodes = sortedNodes(connectorId);
+
+        ConnectorBucketNodeMap connectorBucketNodeMap = getConnectorBucketNodeMap(session, partitioningHandle, sortedNodes);
+
+        NodeSelectionStrategy nodeSelectionStrategy = connectorBucketNodeMap.getNodeSelectionStrategy();
+        switch (nodeSelectionStrategy) {
+            case HARD_AFFINITY:
+            case SOFT_AFFINITY:
+                // todo for soft affinity, we need to add logic if preferDynamic is true
```

Comment:
Since we are already splitting this from the #13966 , let's make sure we don't create a `FixedBucketNodeMap` for `SOFT_AFFINITY` so it would play well with `DynamicLifespanScheduler`, and thus this pull request is self-consistent. Basically we need the functionality in `DynamicBucketNodeMap` to be able to accept a map as initial state and allow further updates to the map as scheduler sees fit.

Reply:
Consider removing TODO here assuming it is supported in the following commit?